### PR TITLE
{math} [gmpich-2017.08] - Add p4est

### PR DESCRIPTION
**Pull request by rubendibattista**

### CODE REVIEW

File: easybuild/easyconfigs/p/p4est/p4est-2.0-gmpich-2017.08.eb [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+toolchain = {'name': 'gmpich', 'version': '2017.08'}
+toolchainopts = {'usempi': True}  
+
+sources = [SOURCELOWER_TAR_GZ]
+source_urls = ['http://p4est.github.io/release/']
+
+configopts = "--enable-mpi"
+
+patches = []
```
Comment (boegel) [inline easybuild/easyconfigs/p/p4est/p4est-2.0-gmpich-2017.08.eb]: @rubendibattista Can be removed?

File: easybuild/easyconfigs/p/p4est/p4est-2.0-gmpich-2017.08.eb [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+homepage = 'http://www.p4est.org'
+description = """p4est is a C library to manage a collection (a forest) of multiple
+connected adaptive quadtrees or octrees in parallel."""
+
+toolchain = {'name': 'gmpich', 'version': '2017.08'}
+toolchainopts = {'usempi': True}  
+
+sources = [SOURCELOWER_TAR_GZ]
+source_urls = ['http://p4est.github.io/release/']
```
Comment (boegel) [inline easybuild/easyconfigs/p/p4est/p4est-2.0-gmpich-2017.08.eb]: @rubendibattista Missing SHA256 checksum (and put `source_urls` above `sources`), `--inject-checksums` will do both for you

File: easybuild/easyconfigs/p/p4est/p4est-2.0-gmpich-2017.08.eb [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+name = 'p4est'
+version = '2.0'
+
+homepage = 'http://www.p4est.org'
+description = """p4est is a C library to manage a collection (a forest) of multiple
+connected adaptive quadtrees or octrees in parallel."""
+
+toolchain = {'name': 'gmpich', 'version': '2017.08'}
+toolchainopts = {'usempi': True}
```
Comment (boegel) [inline easybuild/easyconfigs/p/p4est/p4est-2.0-gmpich-2017.08.eb]: please remove trailing whitespace on this line

File: easybuild/easyconfigs/p/p4est/p4est-2.0-gmpich-2017.08.eb [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+patches = []
+
+dependencies = [('zlib', '1.2.11')]
+
+sanity_check_paths = {
+    'files': ['bin/p4est_simple', 'bin/p4est_step1', 'bin/p4est_step2', 'bin/p4est_step3', 'bin/p4est_step4',
+              'lib/libsc-%(version)s.so', 'lib/libp4est-%(version)s.so', 'lib/libp4est.so'
+              ],
+    'dirs' : []
```
Comment (boegel) [inline easybuild/easyconfigs/p/p4est/p4est-2.0-gmpich-2017.08.eb]: style nitpicking: drop the space before `:`

**Comment by rubendibattista** [comment]
I added OpenBLAS as dependency. I suppose on my machine there's already an implementation of BLAS/LAPACK coming from CentOS repository.

**Comment by boegel** [comment]
@rubendibattista Wouldn't it be better to use a full toolchain then, one that already includes `OpenBLAS` as toolchain component?

**Comment by rubendibattista** [comment]
Using `gmpolf` instead of `gmpich` (w/ deps)

**Comment by boegel** [comment]
@rubendibattista Please open a separate PR for the definition of `gmpolf/2017.08` (and include a HPL easyconfig as test case)

**Comment by boegel** [comment]
replaced with #7816 (with newer toolchain)

**Pull request closed without merging**